commands.go

```diff
@@ -411,7 +411,11 @@ func idCMD(line string, u *user) {
 }
 
 func nickCMD(line string, u *user) {
-	u.pickUsername(line)
+	oldName := u.name
+	showedPrompt, _ := u.pickUsername(line)
+	if showedPrompt {
+		u.room.broadcast(devbot, oldName+" have been renamed to "+u.name+".")
```

Comment:
This should be handled right inside `pickUsername()`. It’s that function’s responsibility to change the username and do whatever is needed to properly change it.

Reply:
The `pickUsername` function already handle the change of the name well. But this shouldn't be its responsibility to broadcast info about the rename as the `pickUsername` is also used for new users where we shouldn't try to broadcast anything in the event of a nickname conflict.

Reply:
Good point

Reply:
We can solve the same way we solved the close-silently issue: make a new function that does the name change and all the prompts quietly, and then make `pickUsername()` a wrapper around that function that checks if the changed name is not the same as `possibleName` and if so, prints the notification. The new user function can then call the quiet function, and all other usages of `pickUsername()` will stay the same. If you agree to this, I can implement it if you'd like.

Reply:
If you want to do that you can go for it or ask me to do it if you prefer. But isn't that somewhat similar to what I already made ? Indeed, I have the `pickUsername()` function that does the quiet processing and the `nickCMD()` function does the loud processing.

Reply:
<!--There are currently three usages of `pickUsername` (nick, new user, room change), one of which should be handled differently from the others. If we make `pickUsername` silent, the other two usages would need to call `nickCMD`.-->

It would not be apparent why other usages call `nickCMD` instead of `pickUsername`, which has a more appropriate name. Also, this could just be me, but calling `cmd` functions in non-cmd contexts feels uncomfortable. 

Another solution is to add a boolean flag specifying whether to be quiet or not, but that seems clunky.

Reply:
I now understand what you mean. I will implement it.

Reply:
It is done.

Reply:
Thanks